src/components/interaction-designer/block-editors/ContactPropertyEditor.vue

```diff
@@ -0,0 +1,143 @@
+            v-model="propertyKey"
+            :label="'flow-builder.contact-property-label' | trans"
+            :placeholder="'flow-builder.enter-contact-property-label' | trans"
+            :valid-state="isValid" />
+        </div>
+      </validation-message>
+
+      <validation-message
+        v-if="propertyAction === PROPERTY_ACTION.SET"
+        #input-control="{ isValid }"
+        :message-key="`block/${block.uuid}/config/set_contact_property/property_value`">
+        <expression-editor
+          :label="'flow-builder.contact-property-expression' | trans"
+          :placeholder="'flow-builder.edit-expression' | trans"
+          :current-expression="propertyValue"
+          :valid-state="isValid"
+          @commitExpressionChange="updatePropertyValue" />
+      </validation-message>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import {IBlock} from '@floip/flow-runner'
+import {Component, Prop} from 'vue-property-decorator'
+import {namespace} from 'vuex-class'
+import Lang from '@/lib/filters/lang'
+import {get} from 'lodash'
+import {mixins} from 'vue-class-component'
+import ValidationMessage from '@/components/common/ValidationMessage.vue'
+import ExpressionEditor from '@/components/common/ExpressionEditor.vue'
+import TextEditor from '@/components/common/TextEditor.vue'
+
+const flowVuexNamespace = namespace('flow')
+
+const NULL_STRING_EXPRESSION = '@(null)'
+const EMPTY_STRING_EXPRESSION = ''
+
+@Component({
+  components: {
+    TextEditor,
+    ExpressionEditor,
+    ValidationMessage,
+  },
+})
+class ContactPropertyEditor extends mixins(Lang) {
+  @Prop() readonly block!: IBlock
+
+  PROPERTY_ACTION = {
+    SET: 'set',
+    CLEAR: 'clear',
+  }
+
+  get propertyAction(): string {
+    if (this.propertyValue === NULL_STRING_EXPRESSION) {
+      return this.PROPERTY_ACTION.CLEAR
+    }
+    return this.PROPERTY_ACTION.SET
+  }
+
+  set propertyAction(value: string) {
+    if (value === this.PROPERTY_ACTION.CLEAR) {
+      this.updatePropertyValue(NULL_STRING_EXPRESSION)
+    } else {
+      this.updatePropertyValue(EMPTY_STRING_EXPRESSION)
+    }
+  }
+
+  get propertyKey(): string {
+    return get(this.block.config, 'set_contact_property.property_key')
+  }
+
+  set propertyKey(value: string) {
+    this.block_updateConfigByPath({
+      blockId: this.block.uuid,
+      path: 'set_contact_property.property_key',
+      value,
+    })
+  }
+
+  get propertyValue(): string {
+    return get(this.block.config, 'set_contact_property.property_value', EMPTY_STRING_EXPRESSION)
+  }
+
+  updatePropertyValue(value: string): void {
+    this.block_updateConfigByPath({
+      blockId: this.block.uuid,
+      path: 'set_contact_property.property_value',
+      value,
+    })
+  }
+
+  @flowVuexNamespace.Mutation block_updateConfigByPath!: (
+    {blockId, path, value}: { blockId: string, path: string, value: string }
+  ) => void
+  @flowVuexNamespace.Mutation block_updateVendorMetadataByPath!: (
```

Comment:
Is this used anywhere? I understood we were going to set the state of the UI - which radio button is selected - in vendor_metadata. I'm not seeing this anywhere though?

Reply:
Good catch here, it's unused, I'm removing it